---
title: "Model AD exercise cohort"
author: "Julio Avelar-Barragan"
date: "2023-03-06"
output: html_document
editor_options: 
  chunk_output_type: console
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)

library(tidyverse)
library(vegan)

setwd("/media/julio/Storage/mAD_exercise/")

#Read in data
merged_species_table = read.table("merged_species_table.tsv", sep = "\t", header = T, check.names = F)
metadata = merged_species_table %>% select(SampleID:Sampletype)
```

```{r Alpha diversity}
#Shannon diversity
Shannon_div = merged_species_table %>% select(!(Genotype:Sampletype)) %>% column_to_rownames(var = "SampleID") %>% 
  diversity(x = ., index = "shannon") %>% data.frame() %>% rename("Shannon" = ".") %>% 
  rownames_to_column(var = "SampleID") %>% left_join(metadata)

#Plot with ggplot
ggplot(data = Shannon_div) +
  aes(x = Genotype, y = Shannon, fill = Exercise) +
  geom_boxplot() +
  geom_point(position = position_jitterdodge()) +
  theme_bw() +
  facet_wrap(~Sex) +
  labs(x = NULL, y = "Shannon Index", fill = "Treatment")
```

Notes:  
* Large difference in the Shannon diversity between males and females, more difference between treatments within genotype.  

```{r Beta diversity}
Beta_MDS = merged_species_table %>% select(!(Genotype:Sampletype)) %>% 
  column_to_rownames(var = "SampleID") %>% metaMDS(comm = .)

Beta_DF = Beta_MDS$points %>% data.frame() %>% rownames_to_column(var = "SampleID") %>% left_join(metadata)

ggplot(data = Beta_DF) +
  aes(x = MDS1, y = MDS2, fill = Sex, pch = Genotype) +
  geom_point(size = 5) +
  geom_text(aes(label = Exercise), color = "black", size = 2) +
  scale_shape_manual(values = c(21,23)) +
  theme_bw() +
  guides(color = "none", fill = guide_legend(override.aes = list(shape = 21))) +
  annotate("text", x = Inf, y = Inf, hjust = 1, vjust = 1, label = bquote("Stress ="~.(round(Beta_MDS$stress, digits = 2))), hjust = 1)

ggplot(data = Beta_DF) +
  aes(x = MDS1, y = MDS2, fill = Exercise, pch = Sex) +
  geom_point(size = 5) +
  geom_text(aes(label = Genotype), color = "black", size = 2) +
  scale_shape_manual(values = c(21,23)) +
  theme_bw() +
  guides(color = "none", fill = guide_legend(override.aes = list(shape = 21))) +
  annotate("text", x = Inf, y = Inf, hjust = 1, vjust = 1, label = bquote("Stress ="~.(round(Beta_MDS$stress, digits = 2))), hjust = 1)

ggplot(data = Beta_DF) +
  aes(x = MDS1, y = MDS2, fill = Genotype, pch = Exercise) +
  geom_point(size = 5) +
  geom_text(aes(label = Sex), color = "black", size = 2) +
  scale_shape_manual(values = c(21,23)) +
  theme_bw() +
  guides(color = "none", fill = guide_legend(override.aes = list(shape = 21))) +
  annotate("text", x = Inf, y = Inf, hjust = 1, vjust = 1, label = bquote("Stress ="~.(round(Beta_MDS$stress, digits = 2))), hjust = 1)
```

```{r Taxonomic profiling, fig.width = 14}
#Taxa bar plot
barplot_df = merged_species_table %>% select(!(Genotype:Sampletype)) %>% column_to_rownames(var = "SampleID") %>%
  decostand(x = ., MARGIN = 1, method = "total") %>% rownames_to_column(var = "SampleID") %>% reshape2::melt() %>% 
  mutate(variable = gsub(".__", "", .$variable)) %>% 
  tidyr::separate(., col = variable, into = c("L1","L2","L3","L4","L5","L6","L7"), sep = "\\|", remove = T, extra = "drop")

#The level of taxonomy you want to look at.  
barplot_df = rename(barplot_df, taxonomy = L7)

#Sum everything by the taxonomic level and replace anything that isn't in the top 10 with 'other'
top_taxa = group_by(barplot_df, taxonomy) %>% summarise(., top_taxa_tmp = sum(value)) %>% arrange(., desc(top_taxa_tmp)) %>% slice(., 1:10)
barplot_df$taxonomy[barplot_df$taxonomy %in% top_taxa$taxonomy != "TRUE"] <- "Other"


barplot_df2 = aggregate(barplot_df$value, by=list(taxonomy=barplot_df$taxonomy, variable = barplot_df$SampleID), FUN=sum) %>%
  merge(., metadata, by.x = "variable", by.y = "SampleID")
barplot_df2 = barplot_df2[order(barplot_df2$taxonomy),] #Re order
barplot_df2 = rbind(barplot_df2[!(barplot_df2$taxonomy == "Other"),],barplot_df2[(barplot_df2$taxonomy == "Other"),]) #Move other to bottom
barplot_df2$taxonomy = factor(barplot_df2$taxonomy, levels = unique(barplot_df2$taxonomy)) #Fix the order

#Custom color pallette.
Julio_color <- c("#003f5c", "#665191", "#d45087", "#ff7c43","#ffa600", "#7F0A57", "#CD9ABB", "#39A9AB", "#71CFC5", "#007947" ,"gray")

ggplot(data = barplot_df2, aes(x = variable, weight = x, fill = taxonomy)) +
  geom_bar(width = 1, color = "black", linewidth = .2) +
  theme_classic() +
  facet_grid(.~Sex+Genotype+Exercise, space = "free", scales = "free") +
  scale_fill_manual(values = Julio_color) +
  theme(axis.text.x = element_text(size = 8, color = "black", angle = 90, hjust = 1, vjust = 0.5), axis.title.y = element_text(size = 16, color = "black"), strip.text.x = element_text(size = 12), strip.background = element_rect(fill="lightblue"), legend.text = element_text(size = 12), legend.title = element_text(size = 16)) +
  labs(x = NULL,
       y = "Relative abundance", fill = "Species")
```

Notes:   
* It looks like the biggest factor associated with microbial composition is sex.  
* The next factor is either genotype or treatment, however, this could be related to mice cage effects.  
* There is an abnormal amount of reads mapping to cyprinid herpesvirus 3, a virus which infects carp fish.  
* Not enough statistical power to reliably determine the effect of each factor on the microbiome.  

```{r Permanova, eval=FALSE, include=FALSE}
adonis2(formula = (merged_species_table %>% select(!c(SampleID:Sampletype))) ~ Exercise, data = merged_species_table, permutations = 999, method = "bray", parallel = 32)
```
